examples/miscellaneous/plot_set_output.py

```diff
@@ -114,3 +115,20 @@
 # This resets `transform_output` to its default value to avoid impacting other
 # examples when generating the scikit-learn documentation
 set_config(transform_output="default")
+
+# %%
+# When configuring the output type with :func:`config_context` the
+# configuration at the time when `transform` or `fit_transform` are
+# called is what counts. Setting these only when you construct or fit
+# the transformer has no effect.
+from sklearn import config_context
+
+scaler = StandardScaler()
+scaler.fit(X_train[num_cols])
+
+with config_context(transform_output="pandas"):
+    # X_test_scaled_pd will contain a dataframe
+    X_test_scaled_pd = scaler.transform(X_test[num_cols])
+
+# X_test_scaled_np will contain a numpy array
+X_test_scaled_np = scaler.transform(X_test[num_cols])
```

Comment:
I would find it better to show the output directly:

```suggestion
# %%
with config_context(transform_output="pandas"):
    scaler.transform(X_test[num_cols])

# %%
scaler.transform(X_test[num_cols])
```

Reply:
Sounds good to me, I want to point out that the suggested change:
 * will not print the pandas output, since it's inside the context manager
 * the numpy output is relatively verbose

See:

<img width="1013" alt="image" src="https://user-images.githubusercontent.com/1419010/210774171-779d6604-e5dc-4165-b0cd-94157420b39e.png">

@glemaitre we could instead do a `head()` of the scaled values for the dataframe, and maybe slice for the numpy array? would that be good?

Reply:
This is how it would look like:

<img width="988" alt="image" src="https://user-images.githubusercontent.com/1419010/210775022-04c4c9da-f03e-4edb-abb4-fd7d2df394db.png">

Reply:
Pushed this ^ change in the latest version 1f4b855

Reply:
Good idea to show the content and the use of `head()`. I like it

Reply:
Yes this would be nice. Thinking about it maybe reverting a small code comment to mention array/dataframe could be better.